src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -663,7 +664,6 @@ private void finish(@NonNull Result r, @CheckForNull Throwable t) {
             onEndBuilding();
         } finally {  // Ensure this is ALWAYS removed from FlowExecutionList
             FlowExecutionList.get().unregister(new Owner(this));
-            listener = null;
```

Comment:
Also need to move up `listener` clearing. Otherwise `listener` wold be non-null (thus still `logUpdated`) at the moment `finish` saved, thus failing `WorkflowRunTest.basics`.